infra-gen2/tool/deploy_gen2.dart

```diff
@@ -0,0 +1,391 @@
+
+    /// create the output file if it does not exist
+    if (!amplifyOutputs.existsSync()) {
+      amplifyOutputs.createSync(recursive: true);
+    }
+
+    print('🏃 Running sandbox deployment for $categoryName');
+    for (final backend in category.backends) {
+      final backendName = backend.name;
+      _deployBackend(
+        category.category,
+        backend,
+        amplifyOutputs.path.replaceFirst('amplify_outputs.dart', ''),
+      );
+
+      /// Skip if there is only one backend
+      if (category.backends.length <= 1) {
+        continue;
+      }
+
+      /// Cache the config contents to create environments map
+      _cacheConfigContents(
+        backendName,
+        amplifyOutputs,
+        amplifyEnvironments,
+      );
+    }
+
+    /// Only append environments if there are multiple backends
+    if (category.backends.length > 1) {
+      _appendEnvironments(
+        amplifyEnvironments,
+        category,
+        amplifyOutputs,
+      );
+    }
+
+    /// Copy amplify_outputs.dart to shared paths
+    _copyAmplifyOutputs(
+      category.sharedOutputs,
+      amplifyOutputs,
+    );
+
+    var bucketName = _createBucketName(categoryName);
+
+    /// Check if the S3 bucket exists
+    final remoteBucketName = _getS3BucketName(bucketName);
+    if (remoteBucketName.isNotEmpty) {
+      bucketName = remoteBucketName;
+      print('🔍 Using existing S3 bucket $bucketName');
+    } else {
+      _createS3Bucket(bucketName);
+    }
+    bucketNames.add(bucketName);
+
+    /// Upload amplify_outputs.dart to S3 bucket
+    _uploadAmplifyOutputs(
+      bucketName,
+      amplifyOutputs.path,
+    );
+
+    print('✅ Deployment for $categoryName Category complete');
+  });
+
+  print('🎉 All backends deployed successfully!');
+
+  print('🪣 S3 Bucket Names: $bucketNames');
+}
+
+/// Deploy Sandbox for a given backend backend
+void _deployBackend(
+  Category category,
+  AmplifyBackend backend,
+  String outputPath,
+) {
+  print(
+      '🏖️  Deploying ${category.name} ${backend.name}, this may take a while...');
+
+  /// Deploy the backend
+  final deployBackend = Process.runSync(
+    'npx',
+    [
+      'ampx',
+      'sandbox',
+      '--outputs-format',
+      'dart',
+      '--outputs-out-dir',
+      outputPath,
+      '--identifier',
+      backend.identifier,
+      '--profile=${Platform.environment['AWS_PROFILE'] ?? 'default'}',
+      '--once',
+    ],
+    workingDirectory: p.join(repoRoot.path, backend.pathToSource),
+    stdoutEncoding: utf8,
+    stderrEncoding: utf8,
+  );
+  if (deployBackend.exitCode != 0) {
+    throw Exception(
+      '❌ Error deploying ${category.name}: '
+      '${deployBackend.stdout}\n${deployBackend.stderr}',
+    );
+  } else {
+    print(
+      '👍 ${category.name} ${backend.identifier} sandbox deployed',
+    );
+  }
+}
+
+/// Cache the config contents to create environments map
+void _cacheConfigContents(
```

Comment:
*suggestion:* Make this a pure function. Instead of mutating `amplifyEnvironments`, have this fn return a new `amplifyEnvironments`.

Reply:
I am not sure if I am following the purpose of this fn. Why not overwrite the file each time? If I have my own config in there will this prevent it from being overwritten?

Reply:
If we performed this each time, we would need to at least add more logic to reason where the new string should appended in the `amplifyEnvironments` map string. This is primarily what I wanted to avoid. 

This approach has the `amplifyEnvironments` map only get created once, after all the back ends have been deployed. Also, I append because we want to preserve the first, historically the default, environment named the `amplifyConfig`. 

To be fair, this logic is easier to see when there are multiple back ends defined in a group.